app/serializers.py

```diff
@@ -0,0 +1,114 @@
+from rest_framework import serializers
+
+from app.models import (
+    Skill, SpokenLanguage, User, Role, Pronoun,
+    LanguageProficiency, SkillProficiency
+)
+
+
+class RoleSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = Role
+        fields = ['role']
+
+
+class SkillProficiencySerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = SkillProficiency
+        fields = ['id', 'level']
+
+
+class RoleSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = Role
+        fields = ['role']
+
+
+class SkillProficiencySerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = SkillProficiency
+        fields = ['id', 'level']
+
+
+class SkillSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = Skill
+        fields = ['id', 'name', 'proficiency']
+
+
+class LanguageProficiencySerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = LanguageProficiency
+        fields = ['id', 'level']
+
+
+class SpokenLanguageSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = SpokenLanguage
+        fields = ['id', 'name', 'proficiency']
+
+
+class PronounSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = Pronoun
+        fields = ['pronoun']
+
+
+class UserSerializer(serializers.ModelSerializer):
+
+    class Meta:
+        model = User
+        fields = [
+            'user_id',
+            'role',
+            'display_name',
+            'about',
+            'avatar',
+            'skills',
+            'pronoun',
+            'spoken_languages',
+            'timezone',
+            'availability'
+        ]
+
+    def create(self, validated_data):
+        skills_data = validated_data.pop('skills')
+        languages_data = validated_data.pop('spoken_languages')
+
+        for skill_data in skills_data:
+            Skill.objects.create(**skill_data)
+        for language_data in languages_data:
+            SpokenLanguage.objects.create(**language_data)
+
+        user = User.objects.create(**validated_data)
+        return user
+
+    def update(self, instance, validated_data):
+
+        role = validated_data.get('role', None)
```

Comment:
The `None` here isn't necessary, I think. Python's `dict.get()` method returns `None` when the key doesn't exist. So this can be `validated_data.get('role')`